web-service-consumer/1.6/modules/ROOT/pages/web-service-consumer-studio.adoc

```diff
@@ -0,0 +1,276 @@
+If the body is not a valid XML, or if the request cannot be created for some reason, you will get a `WSC:BAD_REQUEST` error. If you don't provide body content, the Web Service Consumer Connector attempts to generate one, and this only works for cases where no XML entity is expected in the body. +
+Additionally, some Web Services require that the body has the XML initial declaration declared at the top of it. This tag has the version and encoding information that identifies the document as being XML.
+
+The following XML example shows a DataWeave expression inside the `body` parameter, and how to force the Web Service Connector to dispatch any payload setting the XML initial declaration:
+
+In Studio, click the *Advanced* tab of the Consumer operation and enable the option *Use XML initial declaration* as follows:
+
+.Configure XML Initial Declaration
+image::web-service-consumer-configure-consume.png[Configure XML Initial Declaration]
+
+In the *Configuration XML* tab, the XML looks as follows:
+[source,xml,linenums]
+----
+<wsc:consume config-ref="config" operation="addClients">
+    <wsc:message useXMLInitialDeclaration="true">
+        <wsc:body>
+        #[
+        %dw 2.0
+        output application/xml
+        ns con http://service.soap.clients.namespace/
+        ---
+        con#clients: {
+            client: {
+                name: "admin1",
+                lastname: "textpassword1"
+            },
+            client: {
+                name: "admin2",
+                lastname: "textpassword2"
+            }
+        }]
+        </wsc:body>
+    </wsc:message>
+</wsc:consume>
+----
+
+
+==== Headers Parameter
+
+The `headers` parameter contains application-specific information about the SOAP message, such as authentication, payment, and so on. The parameter is an XML entity, and accepts embedded DataWeave script as its value.
+
+The following XML example shows a DataWeave expression inside the `headers` parameter:
+[source,xml,linenums]
+----
+<wsc:consume config-ref="config" operation="addClients">
+    <wsc:message>
+        <wsc:body>#[payload]</wsc:body>
+        <wsc:headers>
+          #[
+          %dw 2.0
+          output application/xml
+          ns con http://service.soap.clients.namespace/
+          ---
+          "headers": {
+              con#user: "admin",
+              con#pass: "textpassword"
+          }]
+        </wsc:header>
+    </wsc:message>
+</wsc:consume>
+----
+
+==== Attachments Parameter
+
+The `attachments` parameter enables you to bind attachments to the SOAP message. To create attachments for transport over SOAP, declare a DataWeave script in which each entry represents an attachment and its value provides the content of the attachment.
+
+The following XML example shows a DataWeave expression inside the `attachments` parameter, and declares a new attachment called `clientsJson` and its content is stored in the `jsonFile` variable. This variable could be set from a `file:read` operation:
```

Comment:
It's not clear whether the example declares a new attachment or if the DataWeave expression does this. I think the last part of the sentence should be a new sentence. Also which content is stored in the variable? Is it the attachment's content?